File: chattools/src/main/java/aytackydln/chattools/telegram/dto/response/TelegramResponse.java
package aytackydln.chattools.telegram.dto.response;

import aytackydln.chattools.telegram.exception.TelegramException;
import com.fasterxml.jackson.annotation.JsonIgnore;
import org.springframework.lang.Nullable;

public interface TelegramResponse {
    String getMethod();

    @JsonIgnore
    boolean isLimited();

    @Nullable
    TelegramResponse onError(TelegramException e);
    //TODO TelegramResponse onError(Exception e, ServiceContext serviceContext);

    void preSend();
}

File: chattools/src/main/kotlin/aytackydln/chattools/telegram/dto/response/TelegramResponse.kt
package aytackydln.chattools.telegram.dto.response

import aytackydln.chattools.telegram.exception.TelegramException
import com.fasterxml.jackson.annotation.JsonIgnore
import org.springframework.lang.Nullable

interface TelegramResponse {
    val method: String

    @get:JsonIgnore
    val isLimited: Boolean

    @Nullable
    fun onError(e: TelegramException): TelegramResponse?

    //TODO TelegramResponse onError(Exception e, ServiceContext serviceContext);
    fun preSend()
}
